packages/fast-components-react-base/src/viewport-positioner/viewport-positioner.tsx

```diff
@@ -0,0 +1,1011 @@
+                translate = translate < 0 ? 0 : translate + 1;
+                break;
+
+            case ViewportPositionerVerticalPositionLabel.bottom:
+                translate = this.viewportRect.top - this.anchorBottom;
+                translate = translate < 0 ? 0 : translate + 1;
+                break;
+        }
+
+        return translate;
+    };
+
+    /**
+     * Request's an animation frame if there are currently no open animation frame requests
+     */
+    private requestFrame = (): void => {
+        if (this.openRequestAnimationFrame === null && !this.state.disabled) {
+            this.openRequestAnimationFrame = window.requestAnimationFrame(
+                this.updateLayout
+            );
+        }
+    };
+
+    /**
+     * get the anchor element
+     */
+    private getAnchorElement = (): HTMLElement => {
+        if (isNil(this.props.anchor)) {
+            return null;
+        }
+
+        if (this.props.anchor instanceof HTMLElement) {
+            return this.props.anchor;
+        } else {
+            return this.props.anchor.current;
+        }
+    };
+
+    /**
+     * get the viewport element
+     */
+    private getViewportElement = (): HTMLElement => {
+        if (isNil(this.props.viewport)) {
+            return document.scrollingElement as HTMLElement;
+        }
+
+        if (this.props.viewport instanceof HTMLElement) {
+            return this.props.viewport;
+        } else {
+            return this.props.viewport.current;
+        }
+    };
+
+    /**
+     * Converts simple horizontal position to a position label based on AxisPositioningMode
+     */
+    private ConvertHorizontalPositionToLabel = (
+        positioningMode: AxisPositioningMode,
+        position: ViewportPositionerHorizontalPosition
+    ): ViewportPositionerHorizontalPositionLabel => {
+        switch (positioningMode) {
+            case AxisPositioningMode.inset:
+                if (position === ViewportPositionerHorizontalPosition.left) {
+                    return ViewportPositionerHorizontalPositionLabel.insetLeft;
+                } else if (position === ViewportPositionerHorizontalPosition.right) {
+                    return ViewportPositionerHorizontalPositionLabel.insetRight;
+                }
+
+            case AxisPositioningMode.flipOutward:
+                if (position === ViewportPositionerHorizontalPosition.left) {
+                    return ViewportPositionerHorizontalPositionLabel.left;
+                } else if (position === ViewportPositionerHorizontalPosition.right) {
+                    return ViewportPositionerHorizontalPositionLabel.right;
+                }
+            case AxisPositioningMode.uncontrolled:
+                return ViewportPositionerHorizontalPositionLabel.undefined;
+        }
+    };
+
+    /**
+     * Converts simple vertical position to a position label based on AxisPositioningMode
+     */
+    private ConvertVerticalPositionToLabel = (
```

Comment:
the first character of this method should be lowercase, also since this is returning something I would prefer the addition of "get" so "getVerticalPositionToLabel"

Reply:
yep